core/time/shared/tz.rb

```diff
@@ -0,0 +1,39 @@
+describe :time_tz, shared: true do
+  describe "TZ env variable" do
+    before do
+      @tz = ENV["TZ"]
+
+      @get_time = @object
+    end
+
+    after do
+      ENV["TZ"] = @tz
+    end
+
+    it "returns time in timezone specified by TZ env variable" do
+      ENV["TZ"] = "Asia/Hong_Kong"
+      @get_time.call.zone.should == "HKT"
+
+      ENV["TZ"] = "Africa/Dakar"
+      @get_time.call.zone.should == "GMT"
+    end
+
+    it "returns time in UTC if specified incorrect value" do
```

Comment:
This is tricky to spec as it's highly platform-dependent, but this file has some examples which work so far: https://github.com/ruby/spec/blob/7a0aa3e3cad5ee27564a8d42197bf1c8301f4b74/core/time/zone_spec.rb#L65-L89

Reply:
@eregon I see. It's a surprise that time zones "parsing" is platform specific.

I've got "UTC" zone in tests on MacOS locally and unchanged "bla..." zone on TravisCI (I guess on Ubuntu). It's very sad.

Okay. Does it make sense to add these `TZ` related specs at all? As far as there are already very similar specs on `Time#zone`. Actually it was a surprise that TZ env variable affects somehow local timezone at all. I didn't see any mention of this.

Reply:
@eregon I noticed `Time.at` differs a little bit. For some time zones #zone returns different abbreviations e.g. "a/H" instead of "Asi" for Hong Kong.

https://ci.appveyor.com/project/eregon/spec-x948i/builds/21750777/job/x4n78e2jxtbm6rha

Reply:
> Okay. Does it make sense to add these `TZ` related specs at all? As far as there are already very similar specs on `Time#zone`. Actually it was a surprise that TZ env variable affects somehow local timezone at all. I didn't see any mention of this before.

I think we don't need more specs using `TZ`, indeed. Although what you added is fine, but I wouldn't spec specific abbreviations, etc. That's all libc behavior and as you saw they are quite buggy, especially on Windows :(

What's new in 2.6, however is some Time methods accept a "zone object" and that we should spec.

Reply:
https://rubyreferences.github.io/rubychanges/2.6.html#time-support-for-timezones contains quite a bit more information about the new feature.

Reply:
@eregon I understand. I just noticed here https://bugs.ruby-lang.org/issues/14850 example with `TZ` env variable and didn't find any mention in `ruby/spec`

Reply:
RIght, they are all hidden behind the `with_timezone` helper which is meant to change `TZ`.